File: WorldFeed/Server/src/WorldFeed/Client/Requesters/AccountSettingsRequester.cs
namespace WorldFeed.Client.Requesters
{
    using System.Threading.Tasks;

    using WorldFeed.Client;
    using WorldFeed.Common.Client.Validators;
    using WorldFeed.Common.Events;
    using WorldFeed.Common.Public;
    using WorldFeed.Common.Public.Client.Requesters;
    using WorldFeed.Common.Public.Models.Interfaces.DTO;
    using WorldFeed.Common.Public.Parameters.AccountSettingsClient;
    using WorldFeed.Common.Web;
    using WorldFeed.Controllers.AccountSettings;

    public class AccountSettingsRequester : BaseRequester, IAccountSettingsRequester
    {
        private readonly IAccountSettingsController accountSettingsController;
        private readonly IAccountSettingsClientRequiredParametersValidator validator;

        public AccountSettingsRequester(ITwitterClient client, ITwitterClientEvents clientEvents, IAccountSettingsController accountSettingsController,
            IAccountSettingsClientRequiredParametersValidator validator)
        : base(client, clientEvents)
        {
            this.accountSettingsController = accountSettingsController;
            this.validator = validator;
        }

        public Task<ITwitterResult<IAccountSettingsDTO>> GetAccountSettingsAsync(IGetAccountSettingsParameters parameters)
        {
            this.validator.Validate(parameters);
            return ExecuteRequestAsync(request => this.accountSettingsController.GetAccountSettingsAsync(parameters, request));
        }

        public Task<ITwitterResult<IAccountSettingsDTO>> UpdateAccountSettingsAsync(IUpdateAccountSettingsParameters parameters)
        {
            this.validator.Validate(parameters);
            return ExecuteRequestAsync(request => this.accountSettingsController.UpdateAccountSettingsAsync(parameters, request));
        }

        public Task<ITwitterResult<IUserDTO>> UpdateProfileAsync(IUpdateProfileParameters parameters)
        {
            this.validator.Validate(parameters);
            return ExecuteRequestAsync(request => this.accountSettingsController.UpdateProfileAsync(parameters, request));
        }

        public Task<ITwitterResult<IUserDTO>> UpdateProfileImageAsync(IUpdateProfileImageParameters parameters)
        {
            this.validator.Validate(parameters);
            return ExecuteRequestAsync(request => this.accountSettingsController.UpdateProfileImageAsync(parameters, request));
        }

        public Task<ITwitterResult> UpdateProfileBannerAsync(IUpdateProfileBannerParameters parameters)
        {
            this.validator.Validate(parameters);
            return ExecuteRequestAsync(request => this.accountSettingsController.UpdateProfileBannerAsync(parameters, request));
        }

        public Task<ITwitterResult> RemoveProfileBannerAsync(IRemoveProfileBannerParameters parameters)
        {
            this.validator.Validate(parameters);
            return ExecuteRequestAsync(request => this.accountSettingsController.RemoveProfileBannerAsync(parameters, request));
        }
    }
}

File: Sharebook/Client/src/app/sharebook/Client/Requesters/AccountSettingsRequester.ts
import {Inject, Injectable} from "@angular/core";

import {BaseRequester} from "../BaseRequester";
import {IAccountSettingsRequester} from "../../../core/Public/Client/Requesters/IAccountSettingsRequester";
import {ITwitterResult} from "../../../core/Core/Web/TwitterResult";
import {IAccountSettingsController, IAccountSettingsControllerToken} from "../../../controllers/AccountSettings/AccountSettingsController";
import {
  IAccountSettingsClientRequiredParametersValidator,
  IAccountSettingsClientRequiredParametersValidatorToken
} from "../../../core/Core/Client/Validators/AccountSettingsClientRequiredParametersValidator";
import {ITwitterClientEvents, ITwitterClientEventsToken} from "../../../core/Core/Events/TweetinviGlobalEvents";
import {ITwitterClient, ITwitterClientToken} from "../../../core/Public/ITwitterClient";
import {IGetAccountSettingsParameters} from "../../../core/Public/Parameters/AccountSettingsClient/GetAccountSettingsParameters";
import {IAccountSettingsDTO} from "../../../core/Public/Models/Interfaces/DTO/IAccountSettingsDTO";
import {IUpdateAccountSettingsParameters} from "../../../core/Public/Parameters/AccountSettingsClient/UpdateAccountSettingsParameters";
import {IUpdateProfileParameters} from "../../../core/Public/Parameters/AccountSettingsClient/UpdateProfileParameters";
import {IUserDTO} from "../../../core/Public/Models/Interfaces/DTO/IUserDTO";
import {IUpdateProfileImageParameters} from "../../../core/Public/Parameters/AccountSettingsClient/UpdateProfileImageParameters";
import {IUpdateProfileBannerParameters} from "../../../core/Public/Parameters/AccountSettingsClient/UpdateProfileBannerParameters";
import {IRemoveProfileBannerParameters} from "../../../core/Public/Parameters/AccountSettingsClient/RemoveProfileBannerParameters";

@Injectable({
  providedIn: 'root',
})
export class AccountSettingsRequester extends BaseRequester implements IAccountSettingsRequester {
  private readonly _accountSettingsController: IAccountSettingsController;
  private readonly _validator: IAccountSettingsClientRequiredParametersValidator;

  constructor(@Inject(ITwitterClientToken) client: ITwitterClient,
              @Inject(ITwitterClientEventsToken) clientEvents: ITwitterClientEvents,
              @Inject(IAccountSettingsControllerToken) accountSettingsController: IAccountSettingsController,
              @Inject(IAccountSettingsClientRequiredParametersValidatorToken) validator: IAccountSettingsClientRequiredParametersValidator) {
    super(client, clientEvents);
    this._accountSettingsController = accountSettingsController;
    this._validator = validator;
  }

  public getAccountSettingsAsync(parameters: IGetAccountSettingsParameters): Promise<ITwitterResult<IAccountSettingsDTO>> {
    this._validator.validate(parameters);
    return super.executeRequestAsync(request => this._accountSettingsController.getAccountSettingsAsync(parameters, request));
  }

  public updateAccountSettingsAsync(parameters: IUpdateAccountSettingsParameters): Promise<ITwitterResult<IAccountSettingsDTO>> {
    this._validator.validate(parameters);
    return super.executeRequestAsync(request => this._accountSettingsController.updateAccountSettingsAsync(parameters, request));
  }

  public updateProfileAsync(parameters: IUpdateProfileParameters): Promise<ITwitterResult<IUserDTO>> {
    this._validator.validate(parameters);
    return super.executeRequestAsync(request => this._accountSettingsController.updateProfileAsync(parameters, request));
  }

  public updateProfileImageAsync(parameters: IUpdateProfileImageParameters): Promise<ITwitterResult<IUserDTO>> {
    this._validator.validate(parameters);
    return super.executeRequestAsync(request => this._accountSettingsController.updateProfileImageAsync(parameters, request));
  }

  public updateProfileBannerAsync(parameters: IUpdateProfileBannerParameters): Promise<ITwitterResult> {
    this._validator.validate(parameters);
    return super.executeRequestAsync(request => this._accountSettingsController.updateProfileBannerAsync(parameters, request));
  }

  public removeProfileBannerAsync(parameters: IRemoveProfileBannerParameters): Promise<ITwitterResult> {
    this._validator.validate(parameters);
    return super.executeRequestAsync(request => this._accountSettingsController.removeProfileBannerAsync(parameters, request));
  }
}
